jaq-core/src/filter.rs

```diff
@@ -37,6 +37,7 @@ pub enum Ast {
     Comma(Box<Self>, Box<Self>),
     Alt(Box<Self>, Box<Self>),
     Ite(Box<Self>, Box<Self>, Box<Self>),
+    TryCatch(Box<Self>, Option<Box<Self>>),
```

Comment:
Another instance of my lack of experience with large codebases :(

The `empty` filter is defined in `jaq-native`, which imports `jaq-core`. I need to provide `empty` in `lir.rs` in `jaq-core` (as far as I can tell). So that leaves me in a circular imports situation 😬 I can obviously simply provide the inline implementation of `empty`, which would be anti-DRY.

So humm, how would you proceed here? Should I make the `empty` filter an export of `jaq-core` somehow? That also sounds wrong

Reply:
Oh, I forgot that empty was not defined inside the core. My bad.
So we have two options: 
1. we could keep the Option the way you did
2. we could introduce `Ast::empty` the way it is already done with `Ast::recurse0`, simply yielding `[]|.[]`. That way, we could also remove Option from `Ast::Arr` to make construction of non-empty arrays faster, but making empty arrays slower.
What do you think?

Reply:
I'll take a stab at number 2. Thanks!

Reply:
To make things easier for you, I implemented empty in be71aad.